### PR TITLE
fix:modify auto-build.sh bug

### PR DESCRIPTION
**Pull request by Stevent-fei**
### Describe what this PR does / why we need it
https://github.com/sealerio/sealer/issues/1628

### CODE REVIEW

File: context/rootfs/scripts/init-registry.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ certs_dir="$rootfs/certs"
 mkdir -p "$VOLUME" || true
 
 imageServer="nerdctl"
-if [[ $(ls ../cri/docker*.tar.gz) ]]; then
+if [[ $(ls ../cri) == docker*.tar.gz ]]; then
     imageServer="docker"
```
[the remark this answers is not in the document]
Comment (Stevent-fei) [reply context/rootfs/scripts/init-registry.sh]: I think it's OK to compare the directories under CRI. If it is containerd CRI, an internal error message will appear in the script during installation.

**Comment by allencloud** [comment]
@Stevent-fei Could you explain some details about this pull request status? Thanks.

**Comment by Stevent-fei** [comment]
> 

The content of this fix is: when containerd is running, there is an error message in the info when installing the cluster.
image

**Review by kakaZhou719 — APPROVED** [review]

File: context/rootfs/scripts/init-registry.sh [hunk 2]
```diff
@@ -31,55 +31,54 @@ certs_dir="$rootfs/certs"
...
+  imageServer="docker"
 fi
 
 start_registry() {
   n=1
   while ((n <= 3)); do
     echo "attempt to start registry"
-    $imageServer start $container; break || ((n++))
+    $imageServer start $container
+    break || ((n++))
```
Comment (allencloud) [inline context/rootfs/scripts/init-registry.sh:43]: please update here, n++ is redundant.

File: context/rootfs/scripts/init-registry.sh [hunk 3]
```diff
@@ -16,70 +16,118 @@
...
-    (( n++ ))
+  while ((n <= 3)); do
+    registry_status=$(runtimeGetContainerStatus sealer-registry)
+    [[ "$registry_status" == \"running\" ]] && break
+    ((n++))
     sleep 3
   done
+  if [[ "$registry_status" != \"running\" ]]; then
+    echo "sealer-registry is not running, status: $registry_status"
+    return 1
```
Comment (allencloud) [inline context/rootfs/scripts/init-registry.sh:119]: exit 1, rather than return 1.

And make the error info to output into stderr rather stdout.

**Comment by allencloud** [comment]
What is the status of this pull request? @Stevent-fei It is ready for furthure review?

**Comment by allencloud** [comment]
LGTM

**Review by allencloud — APPROVED** [review]

**Pull request merged**